Samples/Server/NativeServer.Tests/NativeServersTests.cpp

```diff
@@ -442,3 +445,81 @@ TEST(BufferCapturerTests, CaptureFrameStereoUsingDirectXBufferCapturer)
 			capturer->staging_frame_buffer_.Get(), 0);
 	}
 }
+
+// --------------------------------------------------------------
+// Decoder tests
+// --------------------------------------------------------------
+TEST(DecoderTests, CanDecodeCorrectly) {
+	//Test implementation initializes decoder implicitly
+	auto h264TestImpl = new H264TestImpl();
+	h264TestImpl->SetEncoderHWEnabled(true);
+	rtc::scoped_refptr<webrtc::VideoFrameBuffer> buffer(
+		h264TestImpl->input_frame_.get()->video_frame_buffer());
+
+	size_t bufferSize = buffer->width() * buffer->height() * 4;
+	size_t RowBytes = buffer->width() * 4;
+	uint8_t* rgbBuffer = new uint8_t[bufferSize];
+
+	// Convert input frame to RGB
+	libyuv::I420ToARGB(buffer->GetI420()->DataY(), buffer->GetI420()->StrideY(),
+		buffer->GetI420()->DataU(), buffer->GetI420()->StrideU(),
+		buffer->GetI420()->DataV(), buffer->GetI420()->StrideV(),
+		rgbBuffer,
+		RowBytes,
+		buffer->width(), buffer->height());
+
+	// Set RGB frame 
+	h264TestImpl->input_frame_.get()->set_frame_buffer(rgbBuffer);
+
+	// Encode frame
+	h264TestImpl->encoder_->Encode(*h264TestImpl->input_frame_, nullptr, nullptr);
+	EncodedImage encodedFrame;
+
+	//Set frame prediction timestamp to 1 (arbitrary value)
+	encodedFrame.prediction_timestamp_ = 1;
+
+	// Extract encoded_frame from the encoder
+	h264TestImpl->WaitForEncodedFrame(&encodedFrame);
+
+	// Check if we have a complete frame with lengh > 0
+	ASSERT_TRUE(encodedFrame._completeFrame);
+	ASSERT_TRUE(encodedFrame._length > 0);
+
+	//The first frame needs to be a keyframe for decoding
```

Comment:
A few more from this line until 515. Add a space after //

Reply:
All done!